### PR TITLE
MRG: Add confidence intervals

### LINKED ISSUE #4423: BUG: Failing dipole test
I think @joewalter mentioned having a failing test in `mne/tests/test_dipole.py` on `master` (right?), and when working on #4355 I came across it, too. This should be fixed in 0.15. I should be able to look soon.

### PR DESCRIPTION
**Pull request by Eric89GXL**
Following the [Xfit manual](http://www.unicog.org/pmwiki/files/MEG/NM20568A-D_SourceMod-5.4.pdf) I think the implementation might be correct. Still need to:

* [x] Add χ^2 and `nfree` like in the C code
* [x] Add tests by cross-checking with Xfit for confidence volume, MNE for `khi2` and `nfree`.
* [x] Add I/O tests for confidence intervals, χ^2, `nfree`.
* [ ] ~~Modify plotting function to account for confidence volume.~~

We can do the plotting modification in a subsequent PR.

Closes #4423.

### CODE REVIEW

**Comment by Eric89GXL** [comment]
Ready for review/merge from my end.

**Comment by agramfort** [comment]
did you compare with xfit results?

**Comment by Eric89GXL** [comment]
Yes, the tests in the code compare to an xfit file

File: mne/dipole.py [hunk 1]
```diff
@@ -166,10 +219,25 @@ def plot_locations(self, trans, subject, subjects_dir=None,
...
+        bgcolor : tuple of length 3
+            Background color in 3D.
+        opacity : float in [0, 1]
+            Opacity of brain mesh.
+        brain_color : tuple of length 3
+            Brain color.
+        fig_name : str
+            Mayavi figure name.
+        fig_size : tuple of length 2
+            Mayavi figure size.
```
Comment (jaeilepp) [inline mne/dipole.py]: What are these?

Comment (jaeilepp) [inline mne/dipole.py]: The conditional on line 292 can also be removed, since ``plot_dipole_locations`` no longer supports plotting with mayavi.

**Comment by Eric89GXL** [comment]
That's rebase error, I'll need to fix it

**Comment by Eric89GXL** [comment]
Okay, rebase error fixed

**Comment by agramfort** [comment]
+1 for MRG when CIs are happy

we'll find a way to visualize these in a subsequent PR

**Comment by agramfort** [comment]
can you just update http://martinos.org/mne/dev/auto_tutorials/plot_dipole_fit.html to demo this new feature?

**Comment by Eric89GXL** [comment]
@agramfort done

**Comment by agramfort** [comment]
I don't see it

**Comment by Eric89GXL** [comment]
Argh, silly OSX ssh-agent grumble grumble. Now I've actually [pushed the commit](https://github.com/mne-tools/mne-python/pull/4355/commits/8062b75e8cbee0f5fff4d07f6efa5982b151a7a4).

**Comment by agramfort** [comment]
cool

merge when CIs are green

**Pull request merged**